trax/data/tf_inputs.py

```diff
@@ -1090,30 +1095,136 @@ def download_and_prepare(dataset_name, data_dir):
   return data_dir
 
 
-def bert_single_sentence_preprocess(batch):
-  for sent1, label in batch:
-    value_vector = np.concatenate(([101], sent1, [102]))
-    segment_embs = np.zeros(sent1.shape[0] + 2, dtype=np.int32)
-    yield value_vector, segment_embs, segment_embs, label, np.int64(1)
+def BertSingleSentenceInputs(batch, labeled=True, CLS_id=101, SEP_id=102):
+  """Prepares inputs for BERT models by adding [SEP] and [CLS] tokens and creating segment embeddings."""
+  if labeled:
+    for sent1, label in batch:
+      value_vector = np.concatenate(([CLS_id], sent1, [SEP_id]))
+      segment_embs = np.zeros(sent1.shape[0] + 2, dtype=np.int32)
+      yield value_vector, segment_embs, segment_embs, label, np.int64(1)
+  else:
+    for sent1, *_ in batch: # row is a tuple with 1 element
+      value_vector = np.concatenate(([CLS_id], sent1, [SEP_id]))
+      segment_embs = np.zeros(sent1.shape[0] + 2, dtype=np.int32)
+      yield value_vector, segment_embs, segment_embs
 
 
-def bert_double_sentence_preprocess(batch):
-  for sent1, sent2, label in batch:
-    value_vector = np.concatenate(([101], sent1, [102], sent2, [102]))
+def BertDoubleSentenceInputs(batch, labeled=True, CLS_id=101, SEP_id=102):
+  """Prepares inputs for BERT models by adding [SEP] and [CLS] tokens and creating segment embeddings."""
+  if labeled:
+    for sent1, sent2, label in batch:
+      value_vector = np.concatenate(([CLS_id], sent1, [SEP_id], sent2, [SEP_id]))
+
+      segment_embs = np.zeros(sent1.shape[0] + sent2.shape[0] + 3, dtype=np.int32)
+      second_sent_start = sent1.shape[0] + 2
+      segment_embs[second_sent_start:] = 1
+      yield value_vector, segment_embs, segment_embs, label, np.int64(1)
+  else:
+    for sent1, sent2 in batch:
+      value_vector = np.concatenate(([CLS_id], sent1, [SEP_id], sent2, [SEP_id]))
 
-    segment_embs = np.zeros(sent1.shape[0] + sent2.shape[0] + 3, dtype=np.int32)
-    second_sent_start = sent1.shape[0] + 2
-    segment_embs[second_sent_start:] = 1
-    yield value_vector, segment_embs, segment_embs, label, np.int64(1)
+      segment_embs = np.zeros(sent1.shape[0] + sent2.shape[0] + 3, dtype=np.int32)
+      second_sent_start = sent1.shape[0] + 2
+      segment_embs[second_sent_start:] = 1
+      yield value_vector, segment_embs, segment_embs
 
 
-def CreateBertInputs(double_sentence=True):  # pylint: disable=invalid-name
-  """Prepares inputs for BERT models by adding [SEP] and [CLS] tokens and creating segment embeddings."""
-  if double_sentence:
-    return bert_double_sentence_preprocess
-  else:
-    return bert_single_sentence_preprocess
+def CreateBertInputs(double_sentence=True, labeled=True, CLS_id=101, SEP_id=101):
+  foo = BertDoubleSentenceInputs if double_sentence else BertSingleSentenceInputs
```

Comment:
s/foo/bert_fn or something else but not foo :-)

Reply:
I changed it to "function". Is that better? :) Just kidding, changed it to bert_inputs_fn